client/client.go

```diff
@@ -0,0 +1,63 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/fabric8-services/fabric8-auth-client/auth"
+	goaclient "github.com/goadesign/goa/client"
+	"github.com/goadesign/goa/middleware/security/jwt"
+	errs "github.com/pkg/errors"
+)
+
+func NewAuthClient(hostURL string) (*AuthClient, error) {
+	u, err := url.Parse(hostURL)
+	if err != nil {
+		return nil, err
+	}
+
+	client := http.Client{}
+	c := auth.New(&doer{
+		target: goaclient.HTTPClientDoer(&client),
+	})
+	c.Host = u.Host
+	c.Scheme = u.Scheme
+	return &AuthClient{c}, nil
+}
+
+type doer struct {
+	target goaclient.Doer
+}
+
+func (d *doer) Do(ctx context.Context, req *http.Request) (*http.Response, error) {
+	token := jwt.ContextJWT(ctx)
+	if token != nil {
+		req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token.Raw))
```

Comment:
currently user_token is set in request but I guess we need some service_acc_token right ?? like [notification_code](https://github.com/fabric8-services/fabric8-notification/blob/master/main.go#L89)